app/models/cta.rb

```diff
@@ -0,0 +1,36 @@
+class CTA < ApplicationRecord
+  CTA_TYPES = { onsite: 1, phone: 2 }
+
+  belongs_to :contact, optional: true
+  belongs_to :location, optional: true
+  belongs_to :call_script, optional: true
+
+  validates_presence_of :title, :description, :website, :start_at, :cta_type
+  validate :validate_unique_cta, on: :create
+
+  scope :upcoming, -> { where("start_at >= ?", Date.today).order("start_at") }
+
+  attr_accessor :cta_type
```

Comment:
Do we still need the `attr_accessor` since you're overriding the getter and setter below?